proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,793 @@
+          "latestVersion": "6.0.0"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "1.1.2",
+          "resolvedVersion": "1.1.2",
+          "latestVersion": "1.1.4"
+        }
+      ]
+    },
+    {
+      "framework": "net5.0",
+      "topLevelPackages": [ 
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278",
+          "latestVersion": "6.0.0"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "1.1.2",
+          "resolvedVersion": "1.1.2",
+          "latestVersion": "1.1.4"
+        }
+      ]
+    }
+  ]
+}
+```
+
+### dotnet list package --deprecated
+
+```dotnetcli
+The following sources were used:
+   https://api.nuget.org/v3/index.json
+   https://apidev.nugettest.org/v3-index/index.json
+
+Project `MyProjectA` has the following deprecated packages
+   [netcoreapp3.1]:
+   Top-level Package                 Requested   Resolved   Reason(s)   Alternative
+   > EntityFramework.MappingAPI      *           6.2.1      Legacy      Z.EntityFramework.Extensions >= 0.0.0
+   > NuGet.Core                      2.13.0      2.13.0     Legacy
+
+Project `MyProjectB` has the following deprecated packages
+   [netcoreapp3.1]:
+   Top-level Package      Requested   Resolved   Reason(s)   Alternative
+   > NuGet.Core           2.13.0      2.13.0     Legacy
+
+   [net5.0]:
+   Top-level Package      Requested   Resolved   Reason(s)   Alternative
+   > NuGet.Core           2.13.0      2.13.0     Legacy
+```
+
+### dotnet list package --deprecated --json
+
+```json
+{
+  "version": 1,
+   "sources": [
+    "https://api.nuget.org/v3/index.json",
+    "https://apidev.nugettest.org/v3-index/index.json"
+  ],
+  "MyProjectA": [
+    {
+      "framework": "netcoreapp3.1",
+      "topLevelPackages": [
+        {
+          "id": "EntityFramework.MappingAPI",
+          "requestedVersion": "*",
+          "resolvedVersion": "6.2.1",
+          "reasons": ["Legacy"],
+          "alternativePackage": {
+            "id": "Z.EntityFramework.Extensions",
+            "versionRange": "[0.0.0,)"
+          }
+        },
+        {
+          "id": "NuGet.Core",
+          "requestedVersion": "2.13.0",
+          "resolvedVersion": "2.13.0",
+          "reasons": ["Legacy"],
+          "alternativePackage": null
+        }
+      ]
+    }
+  ],
+  "MyProjectB": [
+    {
+      "framework": "netcoreapp3.1",
+      "topLevelPackages": [
+        {
+          "id": "NuGet.Core",
+          "requestedVersion": "2.13.0",
+          "resolvedVersion": "2.13.0",
+          "reasons": ["Legacy"],
+          "alternativePackage": null
+        }
+      ]
+    },
+    {
+      "framework": "net5.0",
+      "topLevelPackages": [ 
+        {
+          "id": "NuGet.Core",
+          "requestedVersion": "2.13.0",
+          "resolvedVersion": "2.13.0",
+          "reasons": ["Legacy"],
```

Comment:
I think we should make it this easy to mix with other commands, so maybe we'd call this deprecation reasons? 

what do others think?

Reply:
For me it's reasonable thing to do, changed to `deprecationReasons`

Reply:
Would it maybe be better to group these into a deprecationInfo: {} object or similar? It seems odd to have a few keys that are necessarily related to each other but that aren't grouped into a more meaningful construct that associates them. 

ie:
```diff
{
          "id": "NuGet.Core",
          "requestedVersion": "2.13.0",
          "resolvedVersion": "2.13.0",
-         "reasons": ["Legacy"],
-         "alternativePackage": null
+         "deprecationInfo": {
+           "reasons": ["Legacy"],
+           "alternativePackage": null
+        }
}
```